### PR TITLE
Change public to private in pow.h

### PR DESCRIPTION
**Pull request by mafiya69**

### CODE REVIEW

**Comment by Sumith1896** [comment]
Cool, we eventually should do this for everything.
But more importantly, how do the benchmarks fair?

**Comment by certik** [comment]
+1 as long as benchmarks are not affected. I think they shouldn't be, but it's better to always check.

**Comment by mafiya69** [comment]
Here is the `diff` of output of all benchmarks in benchmarks folder(For a single run)
```
< Without this commit
---
> With this commit

For add1 :
< 497ms
---
> 500ms
For eval_double1 :
< 9369ms
---
> 9414ms
For expand1 :
< 150ms
---
> 131ms
For expand2 :
< 1708ms
---
> 1719ms
For expand2b :
< 118ms
---
> 87ms
For expand3 :
< 24ms
---
> 23ms
For expand6 :
< 126ms
---
> 146ms
For lwbench :
< Time for A : 	     0.003610444
< Time for B : 	     0.001835142
< Time for C : 	     0.012539953
< Time for D : 	     0.000077027
< Time for E : 	     0.000058076
---
> Time for A : 	     0.003631151
> Time for B : 	     0.001774015
> Time for C : 	     0.012386953
> Time for D : 	     0.000080330
> Time for E : 	     0.000059624
For matrix_add1 :
< 9 microseconds
---
> 8 microseconds
For matrix_mul1 :
< 60 microseconds
---
> 65 microseconds
For matrix_mul2 :
< 30 microseconds
---
> 31 microseconds
For symbench :
< Time for R1 : 	     0.000748292
< Time for R2 : 	     0.011396872
< Time for R3 : 	     0.000002151
< Time for R5 : 	     0.000005270
< Time for R7 : 	     0.032147174
< Time for R8 : 	     0.032745271
< Time for S1 : 	     0.023384830
< Time for S2 : 	     0.023592898
< Time for S3 : 	     0.020365235
< Time for S3a : 	     2.349557231
---
> Time for R1 : 	     0.000752161
> Time for R2 : 	     0.011992090
> Time for R3 : 	     0.000002147
> Time for R5 : 	     0.000005046
> Time for R7 : 	     0.031611663
> Time for R8 : 	     0.028062226
> Time for S1 : 	     0.021614932
> Time for S2 : 	     0.023239542
> Time for S3 : 	     0.022386531
> Time for S3a : 	     2.349809748
```

**Comment by certik** [comment]
I think that's fine. +1 to merge. @mafiya69 would you mind resolving the conflicts? Just merge with master and resolve the conflicts. Then we can merge.

**Comment by mafiya69** [comment]
@certik done!

File: symengine/rings.cpp [hunk 1]
```diff
@@ -35,9 +35,9 @@ void expr2poly(const RCP<const Basic> &p, umap_basic_num &syms, umap_vec_mpz &P)
                     }
                 }
             } else if (is_a<Pow>(*p.first)) {
-                RCP<const Basic> sym = rcp_static_cast<const Pow>(p.first)->base_;
-                RCP<const Basic> exp_ = rcp_static_cast<const Pow>(p.first)->exp_;
-                if (not is_a<Integer>(*syms.at(sym)))
+                RCP<const Basic> sym = rcp_static_cast<const Pow>(p.first)->get_base();
+                RCP<const Basic> exp_ = rcp_static_cast<const Pow>(p.first)->get_exp();
+                if (!is_a<Integer>(*syms.at(sym)))
```
Comment (isuruf) [inline symengine/rings.cpp]: ! -> not

File: symengine/pow.cpp [hunk 2]
```diff
@@ -301,16 +301,16 @@ RCP<const Basic> pow_expand(const RCP<const Pow> &self)
         return r;
     }
 
-    if (not  is_a<Integer>(*self->exp_) or not  is_a<Add>(*_base)) {
-        if (neq(*_base, *self->base_)) {
-            return pow(_base, self->exp_);
+    if (! is_a<Integer>(*self->get_exp()) || ! is_a<Add>(*_base)) {
```
Comment (isuruf) [inline symengine/pow.cpp]: ! -> not
|| -> or

**Comment by Sumith1896** [comment]
Looks good

**Comment by isuruf** [comment]
Thanks for the PR

**Pull request merged**